subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/AbstractTestFilteringIntegrationTest.groovy

```diff
@@ -114,15 +145,28 @@ abstract class AbstractTestFilteringIntegrationTest extends AbstractTestingMulti
         """
 
         when:
-        run("test")
+        if (buildSuccess || dryRun) {
+            succeedsWithTestTaskArguments("test")
+        } else {
+            failsWithTestTaskArguments("test")
+        }
 
         then:
         def result = new DefaultTestExecutionResult(testDirectory)
         result.assertTestClassesExecuted("org.gradle.FooTest")
-        result.testClass("org.gradle.FooTest").assertTestsExecuted("pass")
+        result.testClass("org.gradle.FooTest").assertTestOutcomes(testOutcome, testName)
+        if (dryRun) {
+            result.testClassByXml("org.gradle.FooTest").assertStderr(Matchers.emptyString())
+        } else {
+            result.testClassByXml("org.gradle.FooTest").assertStderr(Matchers.containsString("ran FooTest.${testName}!"))
```

Comment:
@octylFractal I believe I've seen that all test was green when you did last checks, but realized that `assertTestCaseStderr` is empty in our test scenario. Could it be a case?

Reply:
I'm not sure what you mean?